k9mail/src/main/java/com/fsck/k9/mailstore/MessageViewInfo.java

```diff
@@ -19,15 +19,15 @@ public MessageViewInfo(List<MessageViewContainer> containers, Message message) {
 
 
     public static class MessageViewContainer {
-        public final String text;
+        public final ViewableContainer viewable;
         public final Part rootPart;
         public final List<AttachmentViewInfo> attachments;
         public final OpenPgpResultAnnotation cryptoAnnotation;
 
 
-        MessageViewContainer(String text, Part rootPart, List<AttachmentViewInfo> attachments,
+        MessageViewContainer(ViewableContainer viewable, Part rootPart, List<AttachmentViewInfo> attachments,
```

Comment:
Replying to @cketti's [comment](https://github.com/k9mail/k-9/pull/1212#issuecomment-215605596):
> Passing a lot of data and a flag through a bunch of layers feels very wrong.

By "a lot of data", do you mean the overhead of the full `viewable` in contrast to `text`?